In [1]:
%load_ext autoreload
%autoreload 2

In [2]:
#Import packages
import pandas as pd
import numpy as np
from joblib import dump

In [3]:
#Load dataset into a dataframe
data = pd.read_csv('../data/raw/GlobalDietaryDatabase_V2.csv')

In [4]:
#Make a copy of the dataframe(df) to be cleaned
df_cleaned = data.copy()

In [5]:
# Define features and target variable
features = df_cleaned.drop(['Unnamed: 0', 'Entity', 'superregion2', 'iso3', 'Year', 'Continent', 'Unit'], axis=1)
target = df_cleaned['Diabetes prevalence (% of population ages 20 to 79)']

In [6]:
from sklearn.model_selection import train_test_split

# Split data into train and test sets
X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, random_state=42)

In [3]:
from sklearn.svm import SVR
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline

# Define the SVR model with a pipeline for scaling
svr_model = make_pipeline(StandardScaler(), SVR(C=1.0, epsilon=0.2))

In [4]:
# Train the model
svr_model.fit(X_train, y_train)

NameError: name 'X_train' is not defined

In [10]:
# Predict on the test set
y_pred = svr_model.predict(X_test)

In [11]:
from sklearn.metrics import mean_squared_error, mean_absolute_error

# Calculate RMSE
rmse = mean_squared_error(y_test, y_pred, squared=False)
print("RMSE:", rmse)

# Calculate MAE
mae = mean_absolute_error(y_test, y_pred)
print("MAE:", mae)

RMSE: 4.0600892072570955
MAE: 2.6562771432870864


RMSE Comparison:
   - Baseline RMSE: 4.411347697591873
   - XGBoost RMSE: 4.049114947585293
   - Random Forest RMSE: 0.304562424258593
   - SVM RMSE: 4.0600892072570955

The SVM model's RMSE falls between the baseline and XGBoost scores, but it's higher than that of the Random Forest model. This indicates that the SVM model's predictions are closer to the actual values than the baseline but not as accurate as the Random Forest model.

MAE Comparison:
   - Baseline MAE: 3.3704969836253955
   - XGBoost MAE: Not provided
   - Random Forest MAE: 0.16713513513513667
   - SVM MAE: 2.6562771432870864

The SVM model performs better than the baseline but is not as accurate as the Random Forest model in terms of both RMSE and MAE. It provides an alternative approach to modeling the diabetes prevalence but doesn't outperform the Random Forest model in this specific task.